orangecontrib/prototypes/widgets/owinteractions.py

```diff
@@ -113,28 +219,41 @@ def compute_score(self, state):
 		removed = self.interaction.removed_h[attr1, attr2] / h
 		return -score, score, gain1, gain2, removed
 
+	def _update_model(self):
+		try:
+			while True:
+				queued = self.add_to_model.get_nowait()
+				row_items = self.row_for_state(queued.score, queued.state)
+				self.rank_model.insertRow(queued.position, row_items)
+		except Empty:
+			pass
+
 	def row_for_state(self, score, state):
 		attrs = sorted((self.attrs[x] for x in state), key=attrgetter("name"))
 		attr_items = []
 		for i, attr in enumerate(attrs):
 			item = QStandardItem(attr.name)
 			item.setData(attrs, self._AttrRole)
 			item.setData(Qt.AlignLeft + Qt.AlignCenter, Qt.TextAlignmentRole)
-			item.setToolTip("{}\nInfo Gain: {:.1f}%".format(attr.name, 100*score[2+i]))
+			item.setToolTip(attr.name)
 			attr_items.append(item)
 		interaction_item = QStandardItem("{:+.1f}%".format(100*score[1]))
 		interaction_item.setData(score[1], self.IntRole)
-		interaction_item.setData(score[4], self.RemovedRole)
+		# arange bars to match columns
+		gains = tuple(map(lambda x: x[1], sorted(enumerate(score[2:4]), key=lambda x: self.attrs[state[x[0]]].name)))
+		interaction_item.setData(gains, self.GainRole)
 		interaction_item.setData(attrs, self._AttrRole)
-		interaction_item.setData(
-			self.NEGATIVE_COLOR if score[1] < 0 else self.POSITIVE_COLOR,
-			gui.TableBarItem.BarColorRole)
-		interaction_item.setToolTip("Entropy removed: {:.1f}%".format(100*score[4]))
 		return [interaction_item] + attr_items
 
 	def check_preconditions(self):
 		return self.master.disc_data is not None
 
+	def stopped(self):
```

Comment:
Does this really fit in stopped? It seems like these things should be set even before stopping.

Reply:
Ok, I see that `OWCorrelations` has the same pattern, so feel free to go forward with the same. But still, do not reimplement the whole function. Instead, call `super().stopped()` and add only the changes.

Reply:
Yes it bugged me also, but I couldn't figure out why it doesn't work if I put it in `__init__` or `initialize`. As far as calling `super().stopped()` it really only saves me 1 line of code so I just skipped it. If that's the accepted practice I will rewrite it though.